Classes/Extensions/DDFileLogger+Buffering.m

```diff
@@ -0,0 +1,165 @@
+// Software License Agreement (BSD License)
+//
+// Copyright (c) 2010-2018, Deusty, LLC
+// All rights reserved.
+//
+// Redistribution and use of this software in source and binary forms,
+// with or without modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the following disclaimer.
+//
+// * Neither the name of Deusty nor the names of its contributors may be used
+//   to endorse or promote products derived from this software without specific
+//   prior written permission of Deusty, LLC.
+
+#import "DDFileLogger+Buffering.h"
+
+@interface DDFileLogger (Internal)
+- (void)logData:(NSData *)data;
+@end
+
+// TODO: Fix it later. ( Protocol adoption? )
```

Comment:
Can we simply put this category into its own file combination (.h/.m) and also implement the `-writeToFile:` method there? This would prevent this TODO...

Reply:
@ffried 
`-writeToFile:` uses `-currentLogFileHandle` which is defined in `DDFileLogger.m`. So, it is impossible to separate them.
Maybe we could add additional protocol or additional method to existing protocol and marked it `optional`(`-logData` will fit it excellently)?